.appveyor.yml

```diff
@@ -17,7 +17,7 @@ install:
   # add python and python user-base to path for pip installs
   - cmd: "C:\\%WINPYTHON%\\python.exe --version"
   - cmd: for /F "tokens=*" %%g in ('C:\\%WINPYTHON%\\python.exe -m site --user-site') do (set PYSITEDIR=%%g)
-  - cmd: "set PATH=C:\\%WINPYTHON%;C:\\%WINPYTHON%\\Scripts;C:\\cygwin64\\bin;C:\\msys64;C:\\ProgramData\\chocolatey\\bin;%PATH%"
+  - cmd: "set PATH=C:\\%WINPYTHON%;C:\\%WINPYTHON%\\Scripts;C:\\MinGW\\bin;C:\\MinGW\\msys\\1.0\\bin;C:\\cygwin\\bin;C:\\ProgramData\\chocolatey\\bin;%PATH%"
```

Comment:
Is msys64 an invalid path?

Reply:
yeah, the standard default path for msys64 is C:\msys64\usr\bin. We should only have 32 bit or 64 bit in the path, but not both.

Reply:
so are we choosing 32 bit then with these changes?
Can you add a more explicit blurb to what path changes you're doing here and why in the summary? It's a bit difficult to tell via the diff and having to mentally reverse engineer the why doesn't make sense..

Reply:
It's a bit complicated why we need 32 bit so bare with me. Looking at the appveyor docs, we should be using 64 bit, because it is on both the 2015 and 2017 appveyor images. 

However, the problem with 64 bit is that the mingw tool will pick up the default 32 bit version first in its list of paths and build with that, and then the Win32/mingw.py test will fail when it tries to execute this:
https://github.com/SCons/scons/blob/9f090132091e56c123262106f3563142787c2fcf/test/Win32/mingw.py#L156
because the TestCmd run process will have the os.environ path which will have 64 bit bins in it:
https://github.com/SCons/scons/blob/9f090132091e56c123262106f3563142787c2fcf/testing/framework/TestCmd.py#L1428-L1433
Using 32 bit it will work, and skip on the 2017 image because the test framework wont find mingw, example recent appveyor 2017 build:
https://ci.appveyor.com/project/SCons/scons/builds/22156672/job/mnvvvpgodtsacjpj#L274

So to summarize we need 32 bit until we have the #3291 implemented and update Win32/mingw.py test to select a version based on whats in the the os.environ PATH.